### PR TITLE
ObjectMapper omits null objects but not empty lists and empty strings.

### PR DESCRIPTION
**Pull request by axthosarouris**
The desired functionality is:
1. Omit from serialization null values
2. Include empty lists
3. Omit non null empty strings.

Due to technical difficulties we cannot omit non null empty strings but we can (as we have done) convert them to nulls when we are accepting them. This means that we will not have empty strings in our system unless we are generating them

### CODE REVIEW

File: core/src/main/java/nva/commons/core/JsonUtils.java [hunk 1]
```diff
@@ -34,19 +33,21 @@ private static ObjectMapper createJsonParser(JsonInclude.Include includeEmptyVal
...
+                .registerModule(new JavaTimeModule())
+                .registerModule(new Jdk8Module())
+                .registerModule(emptyStringAsNullModule())
+                .enable(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT)
+                .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
+                // We want date-time format, not unix timestamps
+                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+                // Ignore null and empty fields
+                .setSerializationInclusion(includeEmptyValuesOption);
```
Comment (brinxmat) [inline core/src/main/java/nva/commons/core/JsonUtils.java:50]: Line 59: is this still true (Corretto)? If so, also include  `//noinspection Convert2Diamond` for the line. It's an annoyance.

File: core/src/test/java/nva/commons/core/JsonUtilsTest.java [hunk 2]
```diff
@@ -214,24 +225,24 @@ public TestObjectForOptionals(@JsonProperty(JSON_KEY) String test) {
 
     private static class TestForEmptyFields {
 
-        @JsonProperty(EMPTY_STRING)
+        @JsonProperty(EMPTY_STRING_FIELD)
```
Comment (brinxmat) [inline core/src/test/java/nva/commons/core/JsonUtilsTest.java:228]: Add     @SuppressWarnings("unused")

Comment (brinxmat) [reply core/src/test/java/nva/commons/core/JsonUtilsTest.java:228]: Also, fix the trailing Greek question mark on line 208

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/DestinationsEventBridgeEventHandlerTest.java [hunk 3]
```diff
@@ -56,7 +55,7 @@ public void handleRequestAcceptsValidEvent() throws JsonProcessingException {
     @Test
     public void handleRequestSerializesObjectsWithoutOmittingEmptyValuesWhenSuchMapperHasBeenSet()
         throws JsonProcessingException, IntrospectionException {
-        final InputStream input = IoUtils.stringToStream(VALID_AWS_EVENT_BRIDGE_EVENT_WITH_EMPTY_VALUES);
+        final InputStream input = IoUtils.stringToStream(VALID_AWS_EVENT_BRIDGE_EVENT);
```
Comment (brinxmat) [inline eventhandlers/src/test/java/no/unit/nva/events/handlers/DestinationsEventBridgeEventHandlerTest.java:58]: Some weird naming in this file, but not your current changes (assertThatProducedJsonContainsOrNotContainsEmptyFields) -> assertHasEmptyFields() and assertDoesNotHaveEmptyFields().

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java [hunk 4]
```diff
@@ -84,36 +83,36 @@ public void handleRequestReThrowsExceptionWhenExceptionIsThrown() {
     @Test
     public void handleRequestSerializesObjectsWithoutOmittingEmptyValuesWhenSuchMapperHasBeenSet()
         throws JsonProcessingException, IntrospectionException {
-        final InputStream input = sampleInputStream(AWS_EVENT_BRIDGE_EVENT_WITH_OBJECT_CONTAINING_EMPTY_VALUES);
+        final InputStream input = sampleInputStream(AWS_EVENT_BRIDGE_EVENT);
         EventHandlerTestClass handler = new EventHandlerTestClass(dtoObjectMapper);
         ObjectNode objectNode = sendEventAndCollectOutputAsJsonObject(input, handler);
-        assertThatJsonObjectContainsOrNotContainsEmptyFields(objectNode, CONTAINS_EMPTY_FIELDS);
+        assertThatJsonObjectContainsOrNotContainsNonNullEmptyFields(objectNode, CONTAINS_EMPTY_FIELDS);
```
Comment (brinxmat) [inline eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java:89]: This should not have a boolean in the sig.

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java [hunk 5]
```diff
@@ -84,36 +83,36 @@ public void handleRequestReThrowsExceptionWhenExceptionIsThrown() {
...
     }
 
-    private void assertThatJsonObjectContainsOrNotContainsEmptyFields(ObjectNode objectNode,
-                                                                      boolean containsEmptyFields)
+    private void assertThatJsonObjectContainsOrNotContainsNonNullEmptyFields(ObjectNode objectNode,
```
Comment (brinxmat) [inline eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java:92]: Shared code for this assert in DestinationEvent…Test

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/SampleEventDetail.java [hunk 6]
```diff
@@ -1,26 +1,42 @@
...
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
 
 public class SampleEventDetail implements WithType {
 
-    private String name;
+    private String emptyString;
```
Comment (brinxmat) [inline eventhandlers/src/test/java/no/unit/nva/events/handlers/SampleEventDetail.java:10]: Tidy the class of unused, or suppress

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java (hunk 4 above)
Comment (axthosarouris) [reply eventhandlers/src/test/java/no/unit/nva/events/handlers/EventHandlerTest.java:89]: fixed

File: eventhandlers/src/test/java/no/unit/nva/events/handlers/SampleEventDetail.java (hunk 6 above)
Comment (axthosarouris) [reply eventhandlers/src/test/java/no/unit/nva/events/handlers/SampleEventDetail.java:11]: fixed

**Pull request merged**